.github/workflows/integrate.yaml

```diff
@@ -46,17 +46,22 @@ jobs:
     runs-on: ubuntu-20.04
     strategy:
       matrix:
-        charm: [controller]
+        charm: [controller, ui]
     steps:
     - name: Check out code
       uses: actions/checkout@v3
     - name: Setup operator environment
-      uses: charmed-kubernetes/actions-operator@main
+      # TODO: Unpin this when https://github.com/charmed-kubernetes/actions-operator/pull/46 is merged
+      uses: claudiubelu/actions-operator@18ebf92ae3043bd3dd15238e5d9b662d7ba08daf
       with:
-        provider: microk8s
-        channel: 1.22/stable
-    - run: sg microk8s -c "microk8s config > ~/.kube/config"
-    - run: sudo snap refresh charmcraft --channel latest/candidate
+          provider: microk8s
+          channel: 1.22/stable
+          charmcraft-channel: latest/candidate
+        # TODO: Unpin this when this bug is resolved: https://bugs.launchpad.net/juju/+bug/1992833.
+        #       In particular, these tests failed deploying the prometheus-k8s charm where it gets an error in
+        #       the "metrics-endpoint-relation-changed" hook.
+          bootstrap-options: --agent-version="2.9.34"
```

Comment:
It seems like the juju team finally has a fix in the 2.9 libjuju branch. Not sure if we want to block this PR on that fix being published, but I would really like us to move away from this workaround soon.